File: java/src/LongestCommonPrefix/LongestCommonPrefix.java
package LongestCommonPrefix;

class LongestCommonPrefix {
    String longestCommonPrefix(String[] strs) {
        StringBuilder prefix = new StringBuilder();
        int charPointer = 0;
        Character currChar;

        if(strs.length == 0) {
            return "";
        }

        outerLoop:
        while(true) {
            currChar = null;
            for(String s : strs) {
                if(charPointer > s.length() - 1) {
                    break outerLoop;
                }

                if(currChar == null) {
                    currChar = s.charAt(charPointer);
                } else if(currChar != s.charAt(charPointer)) {
                    break outerLoop;
                }
            }

            prefix.append(currChar);
            charPointer++;
        }

        return prefix.toString();
    }
}

File: javascript/src/strings/longest_common_prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
  var prefix = "";
  var index = 0;
  var charToCheck;

  if (strs.length === 0 || strs[0] === "") {
    return "";
  } else if (strs.length === 1) {
    return strs[0];
  }

  loop1:
    while (true) {
      if (index >= strs[0].length) {
        break;
      }

      charToCheck = strs[0].charAt(index);
      for (var x = 1; x < strs.length; x++) {
        if (index >= strs[x].length || strs[x].charAt(index) !== charToCheck) {
          break loop1;
        }
      }

      prefix += charToCheck;
      index++;
    }

  return prefix;
};